### PR TITLE
[AIRFLOW-1963] Add config for HiveOperator mapred_queue

### PR DESCRIPTION
**Pull request by edgarRd**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1963

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Previously there's no way to define a default `mapred_queue` for HiveOperators, but there may be valid use cases where a default could be configured in airflow to make HiveOperator tasks more flexible.

Adding configuration setting for specifying a default mapred_queue for
hive jobs using the HiveOperator.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Added tests that check for the default `mapred_queue` and the override via parameters.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

PTAL @aoen @saguziel

### CODE REVIEW

**Comment by edgarRd** [comment]
Failing travis job was a fluke when downloading python: https://travis-ci.org/apache/incubator-airflow/jobs/324683839

File: airflow/config_templates/default_airflow.cfg [hunk 1]
```diff
@@ -170,6 +170,8 @@ default_ram = 512
 default_disk = 512
 default_gpus = 0
 
+# Default mapreduce queue for HiveOperator tasks
+default_hive_mapred_queue =
```
Comment (aoen) [inline airflow/config_templates/default_airflow.cfg]: This is a more generic config section that applies to all operators, this one should be moved to a new [hive] section.

Comment (edgarRd) [reply airflow/config_templates/default_airflow.cfg]: Done, moved it to a `[hive]` section.

File: airflow/operators/hive_operator.py [hunk 2]
```diff
@@ -77,13 +77,16 @@ def __init__(
         self.mapred_queue_priority = mapred_queue_priority
         self.mapred_job_name = mapred_job_name
 
+        # assigned lazily
```
Comment (aoen) [inline airflow/operators/hive_operator.py]: Why is this needed? Might be good to explain in the comment.

Comment (edgarRd) [reply airflow/operators/hive_operator.py]: Just for consistency, as `self.hook` is an attribute assigned outside of init; also, `on_kill` assumes `self.hook` is not `None` and exists. Behavior is not changed, just making the code a bit more consistent. I'll add that in a comment.
